webapp/src/main/java/io/github/microcks/web/SoapController.java

```diff
@@ -393,7 +394,8 @@ private DispatchContext getDispatchCriteriaFromScriptEval(Service service, Strin
          // Evaluating request with script coming from operation dispatcher rules.
          ScriptEngine se = sem.getEngineByExtension("groovy");
          ScriptEngineBinder.bindEnvironment(se, body, requestContext,
-               new ServiceStateStore(serviceStateRepository, service.getId()), request);
+               new ServiceStateStore(serviceStateRepository, service.getId()),
+               HttpHeadersUtil.extractFromHttpServletRequest(request), request);
```

Comment:
See my above comment on `ScriptEngineBinder`. As we're still passing `request`, the binder is still knowledgeable about HTTP stuff, so I would let regular HTTP headers parsing its responsibility.